backends/events/default/default-events.cpp

```diff
@@ -109,6 +110,14 @@ bool DefaultEventManager::pollEvent(Common::Event &event) {
 			// making invalid assumptions about ascii values.
 			event.kbd.ascii = Common::KEYCODE_BACKSPACE;
 			_currentKeyDown.ascii = Common::KEYCODE_BACKSPACE;
+
+		} else if (event.kbd.keycode == Common::KEYCODE_d && (_modifierState & Common::KBD_CTRL)) {
```

Comment:
I would prefer if we did not re-introduce hard-coded key mappings in this class so soon after I made the effort to remove the existing ones :wink:

I guess the way to go to get rid of it to add a new event type `EVENT_DEBUGGER` or something; add an Action binding that event to CTRL+d by default in the default engine keymap (and modify the existing actions in the per-engine overrides); handle that event here.

Reply:
Maybe change the default shortcut to CTRL-SHIFT-D so as to not clash with the SCI games (like QFG games using it as a show-date)

Reply:
Another option is for SCI to define its own engine keymap and bind the debugger to CTRL+SHIFT+d. That way people's habits are not disturbed.

Reply:
I've recently played Loom with my 10 years old son. He used Ctrl-f few times to make things faster and slower, and was frustrated when he accidentally pressed Ctrl-d and got into the debugger.
Since the debugger isn't intended for the users, maybe it makes sense to make it less accessable, and make ctrl-shift-d global default for all engines.
On the other hand, it's difficult to change old habits...

Another idea, maybe have an "easier" default (ctrl-d) for debug binaries, and "complicated" (ctrl-shift-d) default for release binaries.

Reply:
> I guess the way to go to get rid of it to add a new event type `EVENT_DEBUGGER` or something; add an Action binding that event to CTRL+d by default in the default engine keymap (and modify the existing actions in the per-engine overrides); handle that event here.

Not a bad idea, and as suggested, it could be bound to Ctrl-Shift-D to avoid conflicts with any games. I presume that the code needed for the keybinding is in  I'll admit I haven't really been follow the recent keybinding discussions. Is this something already in master, and if so where do I go about setting up the binding.

Reply:
> Is this something already in master, and if so where do I go about setting up the binding.

Yes, keybindings are mostly complete in master. There are essentially two options for the debugger keybinding:
1. We agree to use CTRL+SHIFT+d as the default shortcut for all the engines, and that engines cannot decide to use another default shortcut (some people might not like the furniture being moved around). In that case the keybinding can be defined right in this class, in the `getGlobalKeymap` method.
2. We want engines to be able to set different default keybindings. In that case the keybinding needs to be defined in `MetaEngine::initKeymaps`, and all the per-engine overrides need to be adjusted to account for the new event type.

If you want to catch up on the recent keymapper changes, this page has a bit of documentation: https://wiki.scummvm.org/index.php?title=Keymapper.

Reply:
Thanks. I'll have a review of it tomorrow night. Getting kind of late here.